apps/prairielearn/src/pages/authLogin/authLogin.ts

```diff
@@ -0,0 +1,72 @@
+import { Router } from 'express';
+import asyncHandler = require('express-async-handler');
+import { z } from 'zod';
+import { loadSqlEquiv, queryValidatedRows } from '@prairielearn/postgres';
+
+import { isEnterprise } from '../../lib/license';
+import { AuthLogin, type InstitutionAuthnProvider } from './authLogin.html';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router();
+
+const InstitutionAuthnProviderSchema = z.object({
+  id: z.string(),
+  long_name: z.string(),
+  short_name: z.string(),
+  default_authn_provider_name: z.string(),
+});
+const ServiceSchema = z.string().nullable();
+
+router.get(
+  '/',
+  asyncHandler(async (req, res, _next) => {
+    let institutionAuthnProviders: InstitutionAuthnProvider[] | null = null;
+
+    if (isEnterprise()) {
+      const institutionAuthnProvidersRes = await queryValidatedRows(
+        sql.select_institution_authn_providers,
+        {},
+        InstitutionAuthnProviderSchema
+      );
```

Comment:
A thought I've had: we semi-frequently make queries without any parameters. Would it be worth it to teach this function how to work without the `params` argument? That is, in this case, we'd just have the following:

```suggestion
      const institutionAuthnProvidersRes = await queryValidatedRows(
        sql.select_institution_authn_providers,
        InstitutionAuthnProviderSchema
      );
```

Reply:
I've had the same thought but wasn't sure whether there was a nice way to do this with TS. Would we need to re-order the arguments so the optional one (`params`) comes last?

Reply:
Actually, we would not! With [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) TypeScript can be aware of the difference between the two-argument and three-argument behavior. Then we just check `arguments.length` or the types of the arguments to actually implement that.

Reply:
Oh nice! I didn't think about TS overloads in this context (I haven't actually used them for anything yet). In that case I'm 100% on board. I just added https://github.com/PrairieLearn/PrairieLearn/issues/7739 to summarize our earlier discussion on modernizing the query functions -- that feels like a natural place to make this change as well?

Reply:
Would function overloads also be able to address #7739 in the sense of allowing a `Model` parameter to be optionally used, instead of a separate function for validated rows?